internal/serve/middleware/middleware_test.go

```diff
@@ -136,3 +137,33 @@ func TestSignatureMiddleware(t *testing.T) {
 		assert.JSONEq(t, `{"status":"ok"}`, string(respBody))
 	})
 }
+
+func Test_RecoverHandler(t *testing.T) {
+	// setup logger to assert the logged texts later
+	buf := new(strings.Builder)
+	log.DefaultLogger.SetOutput(buf)
+	log.DefaultLogger.SetLevel(logrus.TraceLevel)
```

Comment:
Shall we use the same as we do in https://github.com/stellar/wallet-backend/blob/a652c1e07264e4503698326bb33038a03474b34d/internal/signing/channel_account_db_signature_client_test.go#L64?